vm/vm.go

```diff
@@ -269,8 +269,20 @@ func (vm *VM) Initialize(
 	vm.builder = builder.NewTime(vm)
 	vm.gossiper = txGossiper
 
-	for _, option := range vm.options {
-		if err := option(vm); err != nil {
+	namespacedConfig := make(map[string]json.RawMessage)
+	if err := json.Unmarshal(controllerConfigBytes, &namespacedConfig); err != nil {
+		return fmt.Errorf("failed to unmarshal namespaced config: %w", err)
+	}
+
+	allocatedNamespaces := make(map[string]struct{})
```

Comment:
nit: this is the idiomatic way to do this in go... but we generally use our implementation of `set.Set` for this in our codebase